CI intermittent nrepl server cli test failures on macos

The macos tests intermittently fail on GH CI

https://github.com/basilisp-lang/basilisp/actions/runs/7596423956/job/20690110714?pr=818
```
=================================== FAILURES ===================================
________________________ TestnREPLServer.test_run_nrepl ________________________

self = <tests.basilisp.cli_test.TestnREPLServer object at 0x10dcb1e50>
run_cli = <function run_cli.<locals>._run_cli at 0x10f0556c0>

    def test_run_nrepl(self, run_cli):
        with tempfile.TemporaryDirectory() as tmpdirpath:
            tmpfilepath = os.path.join(tmpdirpath, ".nrepl-port-test")
            thread = Thread(target=run_nrepl, args=[run_cli, tmpfilepath], daemon=True)
            thread.start()
    
            # give the server some time to settle down.
            #
            # the high retries number is to address the slowness when
            # running on pypy.
            retries = 60
            while not os.path.exists(tmpfilepath):
                time.sleep(1)
                retries -= 1
                assert 0 <= retries
    
            with open(tmpfilepath) as tf:
>               port = int(tf.readline())
E               ValueError: invalid literal for int() with base 10: ''
``` 

It appears that the port file is created by the server but is empty. It might as well be that the server has yielded to the client before writing the port number in it.

PR for possible fix to follow.

--- a/tests/basilisp/cli_test.py
+++ b/tests/basilisp/cli_test.py
@@ -159,7 +159,7 @@ class TestnREPLServer:
             # the high retries number is to address the slowness when
             # running on pypy.
             retries = 60
-            while not os.path.exists(tmpfilepath):
+            while not os.path.exists(tmpfilepath) or os.stat(tmpfilepath).st_size == 0:
                 time.sleep(1)
                 retries -= 1
                 assert 0 <= retries

